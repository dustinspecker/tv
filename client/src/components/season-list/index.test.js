import ConfigContext from '../../config-context'
import SeasonList from '.'
import React from 'react'
import renderer from 'react-test-renderer'
import {StaticRouter} from 'react-router'

test('SeasonList displays list of seasons', () => {
  fetch.mockResponseOnce(
    JSON.stringify({
      seasons: [
        {name: 'Season 1', season: '/show/season1'},
        {name: 'Season 2', season: '/show/season2'}
      ]
    })
  )

  const props = {
    match: {
      params: {
        showname: 'showname'
      }
    }
  }

  const element = (
    <ConfigContext.Provider value={{secureApiServer: 'SECURE_SERVER'}}>
      <StaticRouter context={{}}>
        <SeasonList {...props} />
      </StaticRouter>
    </ConfigContext.Provider>
  )

  const component = renderer.create(element)
  expect(component.toJSON()).toMatchSnapshot()

  return new Promise(resolve => setTimeout(resolve, 0)).then(() => {
    component.update(element)
    expect(component.toJSON()).toMatchSnapshot()
    expect(fetch.mock.calls.length).toEqual(1)
    expect(fetch.mock.calls[0][0]).toEqual('SECURE_SERVER/tv/showname')
  })
})
