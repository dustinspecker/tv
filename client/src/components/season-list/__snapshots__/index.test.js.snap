// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeasonList displays list of seasons 1`] = `
<ul
  className="MuiList-root MuiList-padding"
/>
`;

exports[`SeasonList displays list of seasons 2`] = `
<ul
  className="MuiList-root MuiList-padding"
>
  <a
    href="/show/showname/Season 1"
    onClick={[Function]}
  >
    <li
      className="MuiListItem-root MuiListItem-default MuiListItem-gutters"
      disabled={false}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary"
        >
          Season 1
        </span>
      </div>
    </li>
  </a>
  <a
    href="/show/showname/Season 2"
    onClick={[Function]}
  >
    <li
      className="MuiListItem-root MuiListItem-default MuiListItem-gutters"
      disabled={false}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary"
        >
          Season 2
        </span>
      </div>
    </li>
  </a>
</ul>
`;
